addon/components/mi-date.js

```diff
@@ -1,213 +1,242 @@
-        calendar.hide();
-      }
-      if (isNone(get(this, 'value'))) {
-        $input.inputmask('remove');
-      }
-    } else {
-      this._initDateMask();
-    }
-  }),
-
-  _initDateMask() {
-    let format = this._getDateFormat();
-    let $input = $(this.element).find('.desktop-input');
-    let that = this;
-    $input.inputmask(format.toLowerCase(), {
-      "placeholder": '_',
-      "clearMaskOnLostFocus": true,
-      oncomplete() {
-        that.get('onValueChanged')(that.get('value'));
-      }
-    });
-  },
-
-  isNeverNative: computed('neverNative', function() {
-    return getWithDefault(this, 'neverNative', configuration.getDateConfig().neverNative);
-  }),
-
-  didInsertElement() {
-    if (!isTouchDevice() || this.get('isNeverNative') === true) {
-
-      if (!this.get('disabled')) {
-        this._initDateMask();
-      }
-
-      if (configuration.getDateConfig().useCalendar === true) {
-        this.initPikaday();
-      }
-
-      run.scheduleOnce('afterRender', this, function() {
-        let {
-          calendarButtonClass
-        } = configuration.getDateConfig();
-        set(this, 'calendarClass', calendarButtonClass);
-      });
-    }
-  },
-
-  initPikaday() {
-    let $input = $(this.element).find('.desktop-input');
-    let format = this._getDateFormat();
-
-    let that = this;
-    let pikadayConfig = configuration.getDateConfig();
-    pikadayConfig.onSelect = function(date) {
-      run(function() {
-        set(that, 'value', date);
-        that.onValueChanged(date);
-      });
-    };
-    pikadayConfig.format = format;
-    pikadayConfig.field = $input[0];
-
-    if (this._getShowOn() === 'button') {
-      pikadayConfig.trigger = $(this.element).find('.calendar-button')[0];
-    }
 
-    let options = this.get('options');
-    if (isPresent(options)) {
-      assign(pikadayConfig, options);
-    }
-    set(this, 'pikadayCalendar', new window.Pikaday(pikadayConfig));
+    _getShowOn() {
+        if (configuration.getDateConfig().useCalendar === false) {
+            return 'none';
+        }
+        return getWithDefault(this, 'showOn', configuration.getDateConfig().showOn);
+    },
 
-  },
+    disabledObserver: observer('disabled', function () {
+        let $input = $(this.element).find('.desktop-input');
+
+        if (this.get('disabled')) {
+            let calendar = get(this, 'flatpickrCalendar');
+            if (isPresent(calendar)) {
+                // calendar.hide();
+                calendar.close();
+            }
+            if (isNone(get(this, 'value'))) {
+                $input.inputmask('remove');
+            }
+        } else {
+            this._initDateMask();
+        }
+    }),
+
+    _initDateMask() { //Maska na autocomplete
+        let format = this._getDateFormat();
+        let $input = $(this.element).find('.desktop-input');
+        let that = this;
+        $input.inputmask(format.toLowerCase(), {
+            "placeholder": '_',
```

Comment:
since underscore '_' is used on multiple places and it refers to each other, please create a constatnt named e.g. INPUT_MASK_PLACEHOLDER with value '_' and use it here and on line 194